include/nrsc5.h

```diff
@@ -0,0 +1,173 @@
+#ifndef NRSC5_H_
+#define NRSC5_H_
+
+/*
+ * External API for nrsc5.
+ */
+#include <stdint.h>
+#include <stdlib.h>
+
+/*
+ * Definitions.
+ */
+#define NRSC5_SCAN_BEGIN  87.9e6
+#define NRSC5_SCAN_END   107.9e6
+#define NRSC5_SCAN_SKIP    0.2e6
+
+#define NRSC5_MIME_PRIMARY_IMAGE    0xBE4B7536
+#define NRSC5_MIME_STATION_LOGO     0xD9C72536
+#define NRSC5_MIME_NAVTEQ           0x2D42AC3E
+#define NRSC5_MIME_HERE_TPEG        0x82F03DFC
+#define NRSC5_MIME_HERE_IMAGE       0xB7F03DFC
+#define NRSC5_MIME_HD_TMC           0xEECB55B6
+#define NRSC5_MIME_HDC              0x4DC66C5A
+#define NRSC5_MIME_TEXT             0xBB492AAC
+#define NRSC5_MIME_JPEG             0x1E653E9C
+#define NRSC5_MIME_PNG              0x4F328CA0
+
+/*
+ * Data types.
+ */
+enum
+{
+    NRSC5_SIG_COMPONENT_AUDIO,
+    NRSC5_SIG_COMPONENT_DATA
+};
+
+struct nrsc5_sig_component_t
+{
+    struct nrsc5_sig_component_t *next;
+    uint8_t type;
+    uint8_t id;
+    union
+    {
+        struct {
+            uint16_t port;
+            uint16_t service_data_type;
+            uint8_t type;
+            uint32_t mime;
+        } data;
+        struct {
+            uint8_t port;
+            uint8_t type;
+            uint32_t mime;
+        } audio;
+    };
+};
+typedef struct nrsc5_sig_component_t nrsc5_sig_component_t;
+
+enum
+{
+    NRSC5_SIG_SERVICE_AUDIO,
+    NRSC5_SIG_SERVICE_DATA
+};
+
+struct nrsc5_sig_service_t
+{
+    struct nrsc5_sig_service_t *next;
+    uint8_t type;
+    uint16_t number;
+    const char *name;
+    nrsc5_sig_component_t *components;
+};
+typedef struct nrsc5_sig_service_t nrsc5_sig_service_t;
+
+enum
+{
+    NRSC5_EVENT_LOST_DEVICE,
+    NRSC5_EVENT_IQ,
+    NRSC5_EVENT_SYNC,
+    NRSC5_EVENT_LOST_SYNC,
+    NRSC5_EVENT_MER,
+    NRSC5_EVENT_BER,
+    NRSC5_EVENT_HDC,
+    NRSC5_EVENT_AUDIO,
+    NRSC5_EVENT_SIS,
+    NRSC5_EVENT_ID3,
+    NRSC5_EVENT_SIG,
+    NRSC5_EVENT_LOT,
+    NRSC5_EVENT_TRAFFIC,
+    NRSC5_EVENT_WEATHER
```

Comment:
The traffic and weather events are unused. Did you have plans for them?

Reply:
When I designed the API, it was before I started implementing things. These were non-trivial and not useful enough to warrant the effort. I'll remove the define until we have an implementation.